schema/crdb/dbinit.sql

```diff
@@ -3159,6 +3159,10 @@ CREATE TABLE IF NOT EXISTS omicron.public.bp_target (
     time_made_target TIMESTAMPTZ NOT NULL
 );
 
+CREATE INDEX IF NOT EXISTS lookup_bp_target_by_id ON omicron.public.bp_target (
+    blueprint_id
+);
```

Comment:
I think I would add `version DESC` to the indexed columns.  For the query you're doing, that would allow you to always find the right record quickly, even if the current blueprint had been enabled/disabled a zillion times.  If at some point we wind up querying for the "target" record for a blueprint, we could in principle paginate through this by (blueprint_id, version), although that admittedly seems excessive.

Reply:
We looked at this offline for a while. I claimed (with `EXPLAIN` evidence) that we didn't need to add `version` to the index, which led to "why does CRDB need us to make an index here at all", which led to a very slight query modification that made the index unnecessary: instead of filtering by "the blueprint_id that matches the blueprint_id of the latest target version" (for which CRDB wants an index on `bp_target.blueprint_id`), we can filter by "the version that matches the latest target version" (for which CRDB can use the existing pkey index on `version`). This is implemented in ff89fa5e5